### ISSUE TITLE
Memory leak in v5.4.x

### ISSUE THREAD

**Issue by drakkan** [jackc/pgx#1676]
Hello,

one of the SFTPGo users reported increased memory usage in the latest releases. Memory grows to about 500 MB in less than 8 hours.
The issue is fixed if I [revert pgx](https://github.com/drakkan/sftpgo/commit/abac3cfc8d3170e5ffecfca459e56ebba701ad31) to an older version. In this case the memory usage is constant at less than 100 MB for days as in previous versions.

Profiling outputs.

- In use space after 1 hour

```
Showing top 10 nodes out of 128
      flat  flat%   sum%        cum   cum%
   34.69MB 31.65% 31.65%    34.69MB 31.65%  bytes.growSlice
   21.16MB 19.31% 50.95%    21.16MB 19.31%  github.com/jackc/pgx/v5/internal/iobufpool.init.0.func1
   15.25MB 13.91% 64.87%    15.25MB 13.91%  bufio.NewReaderSize
   11.94MB 10.89% 75.76%    11.94MB 10.89%  bufio.NewWriterSize
       5MB  4.57% 80.32%        5MB  4.57%  crypto/tls.Client
       4MB  3.65% 83.97%        4MB  3.65%  crypto/aes.(*aesCipherGCM).NewGCM
    2.50MB  2.28% 86.25%     2.50MB  2.28%  text/template/parse.(*Tree).newText
       2MB  1.82% 88.08%        2MB  1.82%  crypto/aes.newCipher
    1.50MB  1.37% 89.45%     1.50MB  1.37%  crypto/tls.(*Config).Clone
       1MB  0.91% 90.36%        1MB  0.91%  net.newFD
```

- In use objects after 1 hour

```
Showing top 10 nodes out of 116
      flat  flat%   sum%        cum   cum%
     21845 10.98% 10.98%      21845 10.98%  crypto/tls.(*certCache).active
     18358  9.22% 20.20%      18358  9.22%  text/template/parse.(*Tree).newText
     16385  8.23% 28.43%      16385  8.23%  github.com/jackc/pgx/v5/pgconn/internal/bgreader.New
     16384  8.23% 36.66%      38232 19.21%  crypto/tls.aeadAESGCM
     13108  6.59% 43.25%      13108  6.59%  crypto/aes.(*aesCipherGCM).NewGCM
     13108  6.59% 49.84%      13108  6.59%  time.AfterFunc
     10923  5.49% 55.33%      10923  5.49%  github.com/go-chi/chi/v5.(*node).setEndpoint
     10923  5.49% 60.81%      10923  5.49%  text/template/parse.(*Tree).newField
      9479  4.76% 65.58%       9479  4.76%  bytes.growSlice
      8740  4.39% 69.97%       8740  4.39%  crypto/aes.newCipher
```

- In use space after 3 hours

```
Showing top 10 nodes out of 119
      flat  flat%   sum%        cum   cum%
  102.43MB 40.06% 40.06%   102.43MB 40.06%  bytes.growSlice
   57.45MB 22.46% 62.52%    57.45MB 22.46%  github.com/jackc/pgx/v5/internal/iobufpool.init.0.func1
   19.06MB  7.45% 69.97%    19.06MB  7.45%  bufio.NewWriterSize
   16.52MB  6.46% 76.43%    16.52MB  6.46%  bufio.NewReaderSize
   14.01MB  5.48% 81.91%    14.01MB  5.48%  crypto/tls.Client
    7.50MB  2.93% 84.85%     7.50MB  2.93%  crypto/aes.(*aesCipherGCM).NewGCM
       6MB  2.35% 87.19%        6MB  2.35%  crypto/tls.(*Config).Clone
       6MB  2.35% 89.54%        6MB  2.35%  crypto/aes.newCipher
    3.50MB  1.37% 90.91%     3.50MB  1.37%  github.com/jackc/pgx/v5/pgconn/internal/bgreader.New
       2MB  0.78% 91.69%        2MB  0.78%  text/template/parse.(*Tree).newText
```

- In use objects after 3 hours

```
Showing top 10 nodes out of 121
      flat  flat%   sum%        cum   cum%
     87383 16.31% 16.31%      87383 16.31%  crypto/tls.(*certCache).active
     81922 15.29% 31.60%      81922 15.29%  github.com/jackc/pgx/v5/pgconn/internal/bgreader.New
     32769  6.12% 37.71%      32769  6.12%  crypto/tls.(*Conn).makeClientHello
     32769  6.12% 43.83%      83570 15.60%  crypto/tls.aeadAESGCM
     28297  5.28% 49.11%      28297  5.28%  bytes.growSlice
     26220  4.89% 54.00%      26220  4.89%  crypto/aes.newCipher
     24581  4.59% 58.59%      24581  4.59%  crypto/aes.(*aesCipherGCM).NewGCM
     24577  4.59% 63.18%      52870  9.87%  crypto/tls.(*Conn).readHandshake
     19662  3.67% 66.85%      19662  3.67%  time.AfterFunc
     16396  3.06% 69.91%      16396  3.06%  crypto/tls.Client
```

Unfortunately I cannot reproduce the issue myself, so it is not easy for me to provide a patch. I think the problem may be related to the timer leak reported in #1670

**Comment by jackc** [comment]
I merged #1670 which potentially should resolve this.

**Comment by fuzzmonkey** [comment]
We observed this in production too, reverting to 5.3.1 seems to have fixed the issue. 

@jackc is the fix going to be released soon? Happy to bump up again and confirm.

![Screenshot 2023-07-11 at 13 15 35](https://github.com/jackc/pgx/assets/473265/b2762da0-eabe-4e9b-a294-dda6a48abb9d)
